core/controllers/cron_test.py

```diff
@@ -316,6 +321,149 @@ def _mock_logging_function(msg, *args):
+            'language_code': self.language_code,
+            'content_html': feconf.DEFAULT_INIT_STATE_CONTENT_STR,
+            'translation_html': self.default_translation_html
+        }
+
+        return suggestion_services.create_suggestion(
+            suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, feconf.CURRENT_STATE_SCHEMA_VERSION,
+            self.author_id, add_translation_change_dict,
+            'test description')
+
+    def _assert_reviewable_suggestion_email_infos_are_equal(
+            self, reviewable_suggestion_email_info,
+            expected_reviewable_suggestion_email_info):
+        """Asserts that the reviewable suggestion email info is equal to the
+        expected reviewable suggestion email info.
+        """
+        self.assertEqual(
+            reviewable_suggestion_email_info.suggestion_type,
+            expected_reviewable_suggestion_email_info.suggestion_type)
+        self.assertEqual(
+            reviewable_suggestion_email_info.language_code,
+            expected_reviewable_suggestion_email_info.language_code)
+        self.assertEqual(
+            reviewable_suggestion_email_info.suggestion_content,
+            expected_reviewable_suggestion_email_info.suggestion_content)
+        self.assertEqual(
+            reviewable_suggestion_email_info.submission_datetime,
+            expected_reviewable_suggestion_email_info.submission_datetime)
+
+    def _mock_send_contributor_dashboard_reviewers_emails(
+            self, reviewer_ids, reviewers_suggestion_email_infos):
+        """Mocks
+        email_manager.send_mail_to_notify_contributor_dashboard_reviewers as
+        its not possible to send mail with self.testapp_swap, i.e with the URLs
+        defined in main_cron.
+        """
+        self.reviewer_ids = reviewer_ids
+        self.reviewers_suggestion_email_infos = reviewers_suggestion_email_infos
+
+    def setUp(self):
+        super(
+            CronMailContributorDashboardReviewerOpportunitiesHandlerTests,
+            self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.signup(self.AUTHOR_EMAIL, self.AUTHOR_USERNAME)
+        self.author_id = self.get_user_id_from_email(self.AUTHOR_EMAIL)
+        self.signup(self.REVIEWER_EMAIL, self.REVIEWER_USERNAME)
+        self.reviewer_id = self.get_user_id_from_email(self.REVIEWER_EMAIL)
+        user_services.update_email_preferences(
+            self.reviewer_id, True, False, False, False)
+        self.save_new_valid_exploration(self.target_id, self.author_id)
+
+        self.can_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+        self.cannot_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', False)
+        self.can_send_reviewer_emails = self.swap(
```

Comment:
Instead of swapping, you could also just set the value, e.g. ``config_domain.set_value(committer_id, raw_value)``. This is slightly nicer since it's not mocking out functionality (we should prefer to minimize mocking/faking when real objects are a viable alternative).